protocol-contracts/safe/tasks/verify.ts

```diff
@@ -1,43 +1,52 @@
 import { task } from "hardhat/config";
 
-import { getRequiredEnvVar } from "./utils/loadVariables";
+import { getRequiredEnvVar, getDeployedAddress } from "./utils/loadVariables";
 
 // Verify the SafeL2 contract
 // Example usage:
-// npx hardhat task:verifySafeL2 \
-// --safeL2Address \
-// "0x1234567890123456789012345678901234567890"
-task("task:verifySafeL2")
-  .addParam(
-    "safeL2Address",
-    "address of the already deployed SafeL2 contract that should be verified",
-  )
-  .setAction(async function ({ safeL2Address }, { run }) {
-    await run("verify:verify", {
-      address: safeL2Address,
-      constructorArguments: [],
-    });
+// npx hardhat task:verifySafe --network gateway-testnet
+task("task:verifySafe").setAction(async function (_, { run, network }) {
+  const safeL2Address = await getDeployedAddress(network.name, "SafeL2");
```

Comment:
If you want to fetch addresses from the same network as the task being run on (e.g. `npx hardhat --network gateway-testnet task:verifySafe` and you're looking for deployments on `gateway-testnet`), then use the `deployments` object already available in the hre

```suggestion
task("task:verifySafe").setAction(async function (_, { run, network, deployments }) {
  const safeL2Address = (await deployments.get("SafeL2").address;
```

Reply:
I am not sure about this, current task already working and has been tested on blockscout. Let's keep current version to approve this asap, this optimization can be added later after testing myaybe.